### PR TITLE
[TS] LPS-141200 

### PR DESCRIPTION
**Pull request by rolandpakai**
Hi Balazs,

could you please review my fix?

**Steps to reproduce**

1) Create a Content page, put an Asset publisher on it
2) Configure the AP, so it will be 'Full content' and the default AP
3) Create a web content, put 'EN-title-translated' in the title and content fields
4) Add a French translation, put 'FR-title-translated' in the title and content fields
5) Set the Content page as the display page then publish
6) Create another web content, put 'EN-title-not-translated' in the title and content fields
7) Add a French translation, don't change the title, put 'FR-title-not-translated' in the content field, publish
8) Set the Content page as the display page then publish
9) Create a Widget page, put an Asset Publisher and a Language selector on it
10) Change the language to French
11) Open the 'EN-title-translated' web content
12) You will see the French translation in both the title and content
13) Open the 'FR-title-not-translated' web content

**Current behavior:** You will see the 'EN-title-not-translated' content

**Expected behavior:** You should see the 'FR-title-not-translated' content

**Root cause:**
In case the title has no translation beside the default locale, then no titleMapAsXML parameter with the default locale translation  is sent to the backend for the translations. 
In the reproduction no titleMapAsXML_fr_FR=EN-title-not-translated is sent to the backend.

**Solution:**
On the backend we can retrive this information from the content. 

Thanks, Roland

https://issues.liferay.com/browse/LPS-141200

### CODE REVIEW

**Comment by rolandpakai** [comment]
ci:test:sf

**Comment by rolandpakai** [comment]
ci:test:relevant

**Comment by rolandpakai** [comment]
ci:test:stable

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/impl/JournalArticleLocalServiceImpl.java [hunk 1]
```diff
@@ -9031,6 +9033,33 @@ private int _getUniqueUrlTitleCount(
...
+			String availableLocales = GetterUtil.getString(
+				rootElement.attributeValue("available-locales"));
+			String defaultLocale = GetterUtil.getString(
+				rootElement.attributeValue("default-locale"));
+
+			String[] availableLocalesArray = availableLocales.split(
+				StringPool.COMMA);
+
+			String defaultTitle = urlTitleMap.get(defaultLocale);
+
```
Comment (balazssk) [inline modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/impl/JournalArticleLocalServiceImpl.java:9050]: Is it possible that urlTitleMap.size() > availableLocalesArray.length? If not, maybe we can short circuit the for loop to save time.

**Pull request closed without merging**